File: backend-api-app/src/main/java/no/war/habr/service/dto/UserDto.java
package no.war.habr.service.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import javax.validation.constraints.*;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.Set;

@Builder
@NoArgsConstructor
@AllArgsConstructor
@Getter
public class UserDto {
    private Long id;
    private String username;
    private String firstName;
    private String lastName;
    private String aboutMe;
    private LocalDate birthday;
    private LocalDateTime created;
    private LocalDateTime updated;
    private String condition;
    private Set<String> roles;
}

File: frontend-app/src/app/model/user-dto.ts
export class UserDto {
  constructor(public id: number,
              public username: string,
              public firstName: string,
              public lastName: string,
              public aboutMe: string,
              public birthday: string,
              public created: string,
              public updated: string,
              public condition: string,
              public roles: string[],
  ) {
  }
}
